invokeai/app/services/model_load/model_load_base.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (c) 2024 Lincoln D. Stein and the InvokeAI Team
+"""Base class for model loader."""
+
+from abc import ABC, abstractmethod
+from typing import Optional
+
+from invokeai.app.invocations.baseinvocation import InvocationContext
+from invokeai.backend.model_manager import AnyModel, AnyModelConfig, BaseModelType, ModelType, SubModelType
+from invokeai.backend.model_manager.load import LoadedModel
+from invokeai.backend.model_manager.load.convert_cache import ModelConvertCacheBase
+from invokeai.backend.model_manager.load.model_cache.model_cache_base import ModelCacheBase
+
+
+class ModelLoadServiceBase(ABC):
```

Comment:
Done. `ModelLoadServiceBase` only has the method `load_model(config)` now.

Because this PR isn't merged into #5491, I've put `load_model_by_config()`, `load_model_by_key()` and `load_model_by_attr()` into the `ModelManagerService` class. It does the various lookups needed and then calls the loader service's `load_model()`. When the PR is merged, I'll just remove these and let the API wrapper take care of it.